controlplane/kubeadm/controllers/scale.go

```diff
@@ -90,13 +82,16 @@ func (r *KubeadmControlPlaneReconciler) scaleDownControlPlane(
 		return ctrl.Result{}, errors.Wrapf(err, "failed to create client to workload cluster")
 	}
 
-	// We don't want to health check at the beginning of this method to avoid blocking re-entrancy
-
 	// Wait for any delete in progress to complete before deleting another Machine
 	if controlPlane.HasDeletingMachine() {
 		return ctrl.Result{}, &capierrors.RequeueAfterError{RequeueAfter: deleteRequeueAfter}
 	}
 
+	// We don't want to health check at the beginning of this method to avoid blocking re-entrancy
```

Comment:
```suggestion
```

Reply:
I don't think we should remove this before making sure moving healthcheck at the beginning of this function does not break anything related to machine deletion. We can follow up on this in a new issue.

Reply:
I don't understand, you want to leave the comment as a reminder that if things break it's because of the healthcheck?

Reply:
Either we should move the comment back up there where it was, or remove it altogether